HW_9.py

```diff
@@ -0,0 +1,49 @@
+import argparse
+import csv
+from pymongo import MongoClient
+
+
+def connect_mongo_db(db_name, host='localhost', port=27017, username=None, password=None):
+    if username and password:
+        mongo_uri = 'mongodb://%s:%s@%s:%s/%s' % (username, password, host, port, db)
+        conn = MongoClient(mongo_uri)
+    else:
+        conn = MongoClient(host, port)
+
+    # Remove DB if exists
+    if db_name in conn.list_database_names():
+        conn.drop_database(db_name)
+
+    return conn[db_name]  # Create empty database, return its pointer (object)
+
+
+def create_table_from_csv(csv_file_name, db):
+    table_name = csv_file_name.split('.')[0]  # Get table name from CSV file name
+    with open(csv_file_name, 'r') as f:
+        reader = csv.DictReader(f)
+#         print (reader.fieldnames)
+        mycol = db[table_name]
+        for row in reader:  # insert rows one by one to avoid maximum memory usage (in case of huge CSV file)
+            mycol.insert_one(row)
```

Comment:
A batch of documents can be inserted by passing a list to the **insert_many()** method. PyMongo _will automatically split the batch into smaller sub-batches based on the maximum message size accepted by MongoDB, supporting very large bulk insert operations._

Executing write operations in batches reduces the number of network round trips, increasing write throughput.

Then a code should be changed from:
```
for row in reader:
       mycol.insert_one(row)
```
to
`
mycol.insert_many(reader)
`